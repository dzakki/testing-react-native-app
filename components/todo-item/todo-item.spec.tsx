import { render, screen } from '@testing-library/react-native';
import TodoItem from './index';

describe('TodoItem', () => {
  const props = {
    todo: {
      id: 1,
      title: 'learn typescript',
      completed: true,
    },
    backgroundColor: '#eee',
  };
  test('should render content correctly', () => {
    render(<TodoItem {...props} />);
    expect(screen.getByText(props.todo.title)).toBeVisible();
    expect(screen.getByText(props.todo.title)).toHaveStyle({
      color: 'black',
    });
  });

  test('should match snapshot', () => {
    render(<TodoItem {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  })
});

