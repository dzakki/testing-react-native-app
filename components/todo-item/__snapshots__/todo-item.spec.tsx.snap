// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#eee",
      "borderRadius": 8,
      "marginBottom": 8,
      "padding": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
      }
    }
  >
    learn typescript
  </Text>
</View>
`;
